### ISSUE TITLE
Feature request: Support ES6 Object Property Enhancements

### ISSUE THREAD

**Issue by kphillisjr** [svaarala/duktape#271]
I figured I would create a few Issue statements covering some of the new Features in ECMAScript in a hope that it will make the changes a little easier to handle.  All of this is copied (with slight modifications ) from the "[ECMAScript 6](http://es6-features.org/): New Features: Overview & Comparison" web site ( they are also on [github](https://github.com/lukehoban/es6features/blob/master/README.md)). That said, I will go ahead and get right to the new features...
- [ ] **Property Definition Short Hand**
 ```JavaScript
// New Method:
obj_new_method = { x, y };
// Old method: ECMAScript 5
obj = { x: x, y: y };
```

- [ ] **Simplified Function Definition**
 ```JavaScript
// New Method:
obj_simplified = {
    foo (a, b) { … },
    bar (x, y) { … }
};
// Old Method:
obj = {
    foo: function (a, b) { … },
    bar: function (x, y) { … }
};
```
- [ ] **Computed Property Names** 
 ```JavaScript
// New Method:
obj_new_Method = { foo: "bar", [ "prop_" + foo() ]: 42 };
// Old Method:
obj = { foo: "bar" }; obj[ "prop_" + foo() ] = 42;
```

- [ ] **Advanced Function Definition** - I Think this can be put on hold for now.
 ```JavaScript
// New Method:
obj_simplified = {
    foo (a, b) { … },
    *quux (x, y) { … }
};
// Old Method:
obj = {
    foo: function (a, b) { … }
    /* quux has no equivalent. */
};
```

**Comment by svaarala** [comment]
I'll be doing an ES6 gap analysis later this autumn to get a full list of changes needed. The main thing implementation-wise is to figure out what internal structure needs rework (non-incremental change) to get to ES6. Smaller feature work can then be incremental.

**Comment by svaarala** [comment]
Yeah, there's a need for two kinds of issues: (1) to outline the internal changes and (2) to describe the external changes. Most users are probably interested in specific features in (2), while issues in (1) are preconditions to the features getting implemented. Some overlap between issues is not a problem.

It's probably good to organize the external features by topics like you've done here.
